src/doc/trpl/operators-and-overloading.md

```diff
@@ -1,3 +1,85 @@
 % Operators and Overloading
 
-Coming soon!
+Rust does not have general support for overloading operators or creating new
+operators. However, there are certain operators in Rust that _are_ able to
+be overloaded. To support a particular operator between types, there’s a
+specific trait that you can implement, which then overloads the operator.
+
+For example, the `+` operator can be overloaded with the `Add` trait:
+
+```rust
+use std::ops::Add;
+
+#[derive(Debug)]
+struct Point {
+    x: i32,
+    y: i32,
+}
+
+impl Add for Point {
+    type Output = Point;
+
+    fn add(self, other: Point) -> Point {
+        Point { x: self.x + other.x, y: self.y + other.y }
+    }
+}
+
+fn main() {
+    let p1 = Point { x: 1, y: 0 };
+    let p2 = Point { x: 2, y: 3 };
+
+    let p3 = p1 + p2;
+
+    println!("{:?}", p3);
+}
+```
+
+In `main`, we can use `+` on our two `Point`s, since `we’ve implemented
+`Add<Output=Point>` for `Point`.
+
+All of these traits follow a similar pattern. Let’s look at [`Add`][] in more detail:
+
+```rust
+# mod foo {
+pub trait Add<RHS = Self> {
+    type Output;
+
+    fn add(self, rhs: RHS) -> Self::Output;
+}
+# }
+```
+
+[add]: ../std/ops/trait.Add.html
+
+There’s three types in total involved here: the type you `impl Add` for, and
+two [associated types][at]: `RHS`, which defaults to `Self`, and `Output`. They
+correspond to the three parts of `x + y = z`: the type you `impl` for is `x`,
+`RHS` is `y`, and `Output` is `z`. So this:
```

Comment:
I may just be a little tired in the morning, but I found this a little tough to follow. Perhaps you could start out by saying something like:

> For the expression `let z = x + y`, `x` is the `Self` type, `y` is the `RHS`, and `z` is the `Self::Output` type.

Or something like that? Just wondering if starting out with the three parts of the addition first would help